core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -843,20 +886,23 @@ public void setPinned(LockedInodePath inodePath, boolean pinned, long opTimeMs)
     } else {
       assert inode instanceof InodeDirectory;
       // inode is a directory. Set the pinned state for all children.
-      TempInodePathForDescendant tempInodePath = new TempInodePathForDescendant(inodePath);
-      for (Inode<?> child : ((InodeDirectory) inode).getChildren()) {
-        try {
-          child.lockWriteAndCheckParent(inode);
-        } catch (InvalidPathException e) {
-          // Inode is no longer a child of the directory, continue.
-          continue;
-        }
-        try {
-          tempInodePath.setDescendant(child, getPath(child));
-          setPinned(tempInodePath, pinned, opTimeMs);
-        } finally {
-          child.unlockWrite();
+      try (TempInodePathForDescendant tempInodePath = new TempInodePathForDescendant(inodePath)) {
+        for (Inode<?> child : ((InodeDirectory) inode).getChildren()) {
+          try {
+            child.lockWriteAndCheckParent(inode);
+          } catch (InvalidPathException e) {
+            // Inode is no longer a child of the directory, continue.
+            continue;
+          }
+          try {
+            tempInodePath.setDescendant(child, getPath(child));
+            setPinned(tempInodePath, pinned, opTimeMs);
+          } finally {
+            child.unlockWrite();
+          }
         }
+      } catch (InvalidPathException e) {
+        LOG.warn("setPinned encountered an invalid path {}", inodePath.mUri.getPath());
```

Comment:
When does this happen, and why is this just a warning?

Reply:
This really should not happen. The reason an exception has to be handled is because we changed the constructor of `TempInodePathForDescendant` to throw an exception.  Should we propagate the exception up the call stack?